cmake/flags.cmake

```diff
@@ -108,11 +109,17 @@ function(setupBuildFlags)
       )
 
       set(linux_cxx_link_libraries
-        libc++abi.a
         rt
         dl
       )
 
+      # See osquery/osquery-toolchain#6 for combining c++/c++abi.
+      if(DEFINED OSQUERY_TOOLCHAIN_SYSROOT)
+        list(APPEND linux_cxx_link_libraries
+          c++abi
```

Comment:
This should be still specified as it was previously, because in the presence of a shared libc++abi in the system paths, it will choose that over the static one from the toolchain afaik.
There is a way to force CMake to add to the linker the necessary `-static` flags to require it to search for static only, but then we need to change `rt` and `dl` to `librt.so`, `libdl.so`.

Reply:
I will investigate this, it looks like CMake (the way we have it configured) it defaulting to the static lib, which ends up with duplicate symbols when linking.

Reply:
Ah! I'm of short memory and forgot about this, sorry:
https://github.com/osquery/osquery/blob/8c891ab0a8aa4ac9dce17ce2184cf86eec46fb61/cmake/flags.cmake#L5-L6
So it's indeed already forced to look for static libraries first.
I'm not sure I follow the comment about duplicate symbols? In which case? Which symbols?

Also, to be consistent then, since I failed to do so when I've added those two lines, we should link `rt` and `dl` with their full name.
I've also noticed this:
https://github.com/osquery/osquery/blob/8c891ab0a8aa4ac9dce17ce2184cf86eec46fb61/cmake/flags.cmake#L106
Which it would be better to be moved in `linux_cxx_link_libraries`, using again its full name, since it's shared.